src/keys/get-management-token.ts

```diff
@@ -67,25 +70,45 @@ const getTokenFromOneTimeToken = async (
     `Successfully retrieved CMA Token for app ${appInstallationId} in space ${spaceId} and environment ${environmentId}`
   )
 
-  return JSON.parse(response.body).token
+  return JSON.parse(response.body).token as Promise<string>
 }
 
 /**
  * Factory method for GetManagementToken
  * @internal
  */
 export const createGetManagementToken = (log: Logger, http: HttpClient) => {
-  return (privateKey: unknown, opts: GetManagementTokenOptions): Promise<string> => {
+  return async (privateKey: unknown, opts: GetManagementTokenOptions): Promise<string> => {
     if (!(typeof privateKey === 'string')) {
       throw new ReferenceError('Invalid privateKey: expected a string representing a private key')
     }
 
+    const cacheKey = opts.appInstallationId + opts.environmentId + privateKey
+    if (opts.reuseToken !== false) {
+      const existing = cache.get(cacheKey) as string
+      if (existing !== undefined) {
+        return existing
+      }
+    }
+
     const appToken = generateOneTimeToken(
       privateKey,
       { appId: opts.appInstallationId, keyId: opts.keyId },
       { log }
     )
-    return getTokenFromOneTimeToken(appToken, opts, { log, http })
+    const ott = await getTokenFromOneTimeToken(appToken, opts, { log, http })
+
+    if (opts.reuseToken !== false) {
```

Comment:
What about?
```suggestion
    if (opts.reuseToken) {
```

Suggesting it because, if we're really concerned about the shape of the option object in this deep detail, maybe we should be validating it as soon as it gets in (maybe replacing [this ugly](https://github.com/contentful/node-apps-toolkit/pull/10/files#diff-7780f384417135af6236f8c934c39d9e6ad1f24ae9a666c479c16263af2ec48cR82) statement I have written a couple of weeks back). With that said, I personally don't think params validation is strictly necessary for now

Reply:
The problem here is that we want the default behavior to be `reuseToken: true` so I only checked if the user might have explicitly set it to `false`.
Doing `cacheToken = opts.reuseToken !== false ? true : false` seemed worse.
```
if (opts.reuseToken === undefined) {
  opts.reuseToken = true
}
```
would also work.